src/ai/backend/manager/api/stream.py

```diff
@@ -626,36 +627,43 @@ async def stream_conn_tracker_gc(root_ctx: RootContext, app_ctx: PrivateContext)
     shared_config: SharedConfig = root_ctx.shared_config
     try:
         while True:
-            no_packet_timeout: timedelta = tx.TimeDuration().check(
-                await shared_config.etcd.get('config/idle/app-streaming-packet-timeout') or '5m',
-            )
-            async with app_ctx.conn_tracker_lock:
-                now = await redis.execute(redis_live, lambda r: r.time())
-                now = now[0] + (now[1] / (10**6))
-                for session_id in app_ctx.active_session_ids.keys():
-                    conn_tracker_key = f"session.{session_id}.active_app_connections"
-                    prev_remaining_count = await redis.execute(
-                        redis_live,
-                        lambda r: r.zcount(conn_tracker_key, float('-inf'), float('+inf')),
-                    )
-                    removed_count = await redis.execute(
-                        redis_live,
-                        lambda r: r.zremrangebyscore(
-                            conn_tracker_key, float('-inf'), now - no_packet_timeout.total_seconds(),
-                        ),
-                    )
-                    remaining_count = await redis.execute(
-                        redis_live,
-                        lambda r: r.zcount(conn_tracker_key, float('-inf'), float('+inf')),
-                    )
-                    log.debug(f"conn_tracker: gc {session_id} "
-                              f"removed/remaining = {removed_count}/{remaining_count}")
-                    if prev_remaining_count > 0 and remaining_count == 0:
-                        await root_ctx.idle_checker_host.update_app_streaming_status(
-                            session_id,
-                            AppStreamingStatus.NO_ACTIVE_CONNECTIONS,
+            try:
+                no_packet_timeout: timedelta = tx.TimeDuration().check(
+                    await shared_config.etcd.get('config/idle/app-streaming-packet-timeout') or '5m',
+                )
```

Comment:
The outer try-catch block should embrace this part only if there no etcd access below to minimize the change.